src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/magical/MagicalZombiePills.java

```diff
@@ -47,7 +51,8 @@ public EntityInteractHandler getItemHandler() {
         return (e, item, offhand) -> {
             Entity entity = e.getRightClicked();
 
-            if (entity.getType() == EntityType.ZOMBIE_VILLAGER) {
+            if (SlimefunPlugin.getMinecraftVersion().isAtLeast(MinecraftVersion.MINECRAFT_1_16) && entity instanceof PigZombie
+                    || entity instanceof ZombieVillager) {
                 Player p = e.getPlayer();
```

Comment:
What if you put Line 56-62 into a seperate method like `useItem` or whatever and then just call this method before the specific heal method, this way we save redundancy even further because we no longer need two different if statements for the same check.